### ISSUE TITLE
Cannot find name 'Materialize' with ng build (angular cli)

### ISSUE THREAD

**Issue by Jesse-longname** [InfomediaLtd/angular2-materialize#287]
I created a new project using `ng new PROJECT_NAME` and followed the angular-cli installation instructions. I have `Materialize.updateTextFields()` in my app.component.ts. When I run `ng build`, it results in an Error `Error in [path]: Cannot find name 'Materialize'.`
Any ideas on how to resolve this? 

package.json:
```
{
  "name": "something",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^4.0.0",
    "@angular/common": "^4.0.0",
    "@angular/compiler": "^4.0.0",
    "@angular/core": "^4.0.0",
    "@angular/forms": "^4.0.0",
    "@angular/http": "^4.0.0",
    "@angular/platform-browser": "^4.0.0",
    "@angular/platform-browser-dynamic": "^4.0.0",
    "@angular/router": "^4.0.0",
    "angular2-materialize": "^15.0.4",
    "core-js": "^2.4.1",
    "hammerjs": "^2.0.8",
    "jquery": "^2.2.4",
    "materialize-css": "^0.98.2",
    "rxjs": "^5.1.0",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "@angular/cli": "1.1.0",
    "@angular/compiler-cli": "^4.0.0",
    "@angular/language-service": "^4.0.0",
    "@types/jasmine": "2.5.45",
    "@types/materialize-css": "^0.98.1",
    "@types/node": "~6.0.60",
    "codelyzer": "~3.0.1",
    "jasmine-core": "~2.6.2",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.1.1",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.2",
    "ts-node": "~3.0.4",
    "tslint": "~5.3.2",
    "typescript": "~2.3.3"
  }
}
```

**Comment by rubyboy** [comment]
@Jesse-longname this is related to missconfiguration of the project. Please make sure you follow the CLI instructions (https://github.com/InfomediaLtd/angular2-materialize#installing--configuring-angular2-materialize-in-projects-created-with-the-angular-cli) and update the ,angular-cli.json properly.
If it's still not working, please share the repo here and someone could take a look and help configure it to work with angular2-materialize.

**Comment by markbudiak** [comment]
@rubyboy I have the same problem. I followed all the instructions but I am still getting this error:
`ERROR in [path]: error TS2304: Cannot find name 'Materialize'.`

This is my angular-cli configuration:
{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "project": {
    "name": "job-service-ui"
  },
  "apps": [
    {
      "root": "src",
      "outDir": "dist",
      "assets": [
        "assets",
        "favicon.ico"
      ],
      "index": "index.html",
      "main": "main.ts",
      "polyfills": "polyfills.ts",
      "test": "test.ts",
      "tsconfig": "tsconfig.app.json",
      "testTsconfig": "tsconfig.spec.json",
      "prefix": "app",
      "styles": [
        "../node_modules/materialize-css/dist/css/materialize.css",
        "styles.css"
      ],
      "scripts": [
        "../node_modules/jquery/dist/jquery.min.js",
        "../node_modules/hammerjs/hammer.js",
        "../node_modules/materialize-css/dist/js/materialize.js"
      ],
      "environmentSource": "environments/environment.ts",
      "environments": {
        "dev": "environments/environment.ts",
        "prod": "environments/environment.prod.ts"
      }
    }
  ],
  "e2e": {
    "protractor": {
      "config": "./protractor.conf.js"
    }
  },
  "lint": [
    {
      "project": "src/tsconfig.app.json",
      "exclude": "**/node_modules/**"
    },
    {
      "project": "src/tsconfig.spec.json",
      "exclude": "**/node_modules/**"
    },
    {
      "project": "e2e/tsconfig.e2e.json",
      "exclude": "**/node_modules/**"
    }
  ],
  "test": {
    "karma": {
      "config": "./karma.conf.js"
    }
  },
  "defaults": {
    "styleExt": "css",
    "component": {}
  }
}

I am using angular5, could this problem be related to this fact? How can I resolve this issue?
Sorry but I cannot share the whole project.

**Comment by rajatrawataku1** [comment]
@rubyboy  did you get the solution for the above problem?
I have tried to import Materialise in my typescript file and then use  Materialize.toast('Published !!!! ',4000);
this works now. But my main problem is if I try to close my modal programmatically I am not able to close it.

I am trying this command:
$('#modal2').modal('close');

I am getting this error:
Property 'modal' does not exist on type 'JQuery<HTMLElement>'.

I have imported two files like this 
import * as Materialize from 'materialize-css';
import * as $ from 'jquery';

Any help would be highly appreciated.
